activejdbc/src/main/java/org/javalite/activejdbc/Model.java

```diff
@@ -249,6 +261,79 @@ protected void hydrate(Map<String, Object> attributesMap, boolean fireAfterLoad)
             fireAfterLoad();
         }
 
+        return changedAttributeNames;
+    }
+
+    /**
+     * Verifies if the passed value for attributeName would set this instance to modified state.
+     * @param attributeName
+     * @param newValue
+     * @return
+     */
+    protected boolean wouldAttributeValueModifyModel(String attributeName, Object newValue) {
+
+        Object currentAttributeValue = get(attributeName);
+
+        if (isNew() || (currentAttributeValue == null && newValue != null))
+            return true;
+
+        if (currentAttributeValue != null) {
+
+            if (newValue == null)
+                return true;
+
+            if (newValue.getClass() != currentAttributeValue.getClass()) {
+                Object convertedObject = convertValueToTargetClass(newValue, currentAttributeValue);
+                if (convertedObject != null) {
+                    return !convertedObject.equals(currentAttributeValue);
+                } else
+                    return true;
+            } else {
+                return !newValue.equals(currentAttributeValue);
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Tries to convert the given new attribute value type to the one of the existing attribute value.
+     * @param newAttributeValue
+     * @param existingAttributeValue
+     * @return
+     */
+    protected Object convertValueToTargetClass(Object newAttributeValue, Object existingAttributeValue) {
```

Comment:
I see what you are doing here, and code looks fine. However, this is a more philosophical issue. I can stick anything I want  into the model on the one hand, and the database driver can stick anything that comes from the database into the same model. The two values may or may not be the same type. Now, this code assumes that the existing value has a "right of way", that is, whichever value type is set on the model first, is correct. However, this may or may not be correct. 

Additionally, if you are supporting all these types, then you need to support them all: 
https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html

This is exactly the rabbit hole I wanted to avoid. 

Also, the `convertedObject` on line 286 is a throw-away object, and depending on the number of attributes/columns and weight of these objects, you might be trashing the heap space if this is run in a tight loop over a large resultset.

So... the goal was optimization, right? In other words, if there are no changes, why save a model? However, with this approach you might cause side effects that may be worse. 

Don't get me wrong, I welcome suggestions and value your effort. Also, you write clean code. However, I have my reservations.

Reply:
Adding to that, the framework as it is today is a complete pass-through: http://javalite.io/pass_through_framework. As soon as you start adding any type-related intelligence, this will be lost and eventually will turn into Hibernate :)

Reply:
Thanks for the review! I see and can somehow understand your concerns. Let me clarify some aspects:

* i always looked at this pull request as a starting point for the topic, therefore possibly not being complete or someone else could have better (or nearer to the pass-through idea) ideas for that feature
* since i use your own convert methods (which you use as well) i cannot see other side effects which could already happen in your existing code as well. The worst thing i see is that an attribute value cannot be converted and throws an exception

I still think this topic should be solved somehow (perhaps not with that pull request) because (in my opinion) it's wrong when only the field 'updated_at' is actually updated since all other attribute values stay the same, where Model.isModified() cannot distinguish between same and actually changed attribute values.

Reply:
@rivella50 I totally agree that we need to consider ways to solve this to improve performance. However, there are concerns I outlined above. I think the major issue with this is inability to compare attribute values without data conversion. How about this rule: lets simply do this: `newVal.equals(oldVal)` - without regard to the type. The developer should be responsible for type conversion if they set something to the model. In the worst case, it will simply save more data to the DB. 

In other words, the developer should know what Java type an attribute is, and provide that type. This way, we can keep the "pass-through" model, and if someone wants to keep models from saving the same data, they will need to be careful with types. 

As far as the `Convert` class, this is just a Swiss Army converter. It was not built to cover all the JDBC types.

Reply:
Ok, let's try it. The question is: How often the Converter actually had to convert the initial (old) value to a different type before it was saved and therefore your new proposal then would lead to an (unnecessary) update. But it's worth a try.

Reply:
Well, my proposal was to not do any conversions at all.

Reply:
Ah, sorry, i've mixed it up with the converters a user can register for attributes in his models.
But with that update values of course also will be converted.

Reply:
I'm in vacation right now for one week, therefore not near my dev system.
If you have time to change and retest the method wouldAttributeValueModifyModel()
please have a go, else i can do it by the end of next week.

Reply:
I'm buried in other tasks, main  job, broken build, etc. Have a great vacation - this is not urgent by any means:)